docs/ui-toolkit/components/_posts/1970-01-06-GridView.md

```diff
@@ -41,91 +41,99 @@ The main idea behind this approach is to allow developers to have a variable num
-        </TouchableOpacity>
-      );
-    }
-
-    const cellViews = rowData.map((restaurant, id) => {
-    return (
-        <TouchableOpacity key={id} styleName="flexible">
-          <Card styleName="flexible">
-            <Image
-              styleName="medium-wide"
-              source={% raw %}{{{% endraw %} uri: restaurant.image.url  }}
-            />
-            <View styleName="content">
-              <Subtitle numberOfLines={3}>{restaurant.name}</Subtitle>
-              <View styleName="horizontal">
-                <Caption styleName="collapsible" numberOfLines={2}>{restaurant.address}</Caption>
-              </View>
-            </View>
-          </Card>
-        </TouchableOpacity>
-      );
-    });
+renderRow(rowData, sectionId, index) {
+  // rowData contains grouped data for one row,
+  // so we need to remap it into cells and pass to GridRow
+  if (index === '0') {
     return (
-      <GridRow columns={2}>
-        {cellViews}
-      </GridRow>
+      <TouchableOpacity key={index}>
+        <ImageBackground
+          styleName="large"
+          source={% raw %}{{{% endraw %} uri: rowData[0].image.url }}
+        >
+          <Tile>
+            <Title styleName="md-gutter-bottom">{rowData[0].name}</Title>
+            <Subtitle styleName="sm-gutter-horizontal">{rowData[0].address}</Subtitle>
+          </Tile>
+        </ImageBackground>
+        <Divider styleName="line" />
+      </TouchableOpacity>
     );
   }
 
-  render() {
-    // Group the restaurants into rows with 2 columns, except for the
-    // first article. The first article is treated as a featured article
-    let isFirstArticle = true;
-    const groupedData = GridRow.groupByRows(this.state.restaurants, 2, () => {
-      if (isFirstArticle) {
-        isFirstArticle = false;
-        return 2;
-      }
-
-      return 1;
-    });
+  const cellViews = rowData.map((restaurant, id) => {
     return (
-      <Screen>
-        <ListView
-          data={groupedData}
-          renderRow={this.renderRow}
-        />
-      </Screen>
+      <TouchableOpacity key={id} styleName="flexible">
+        <Card styleName="flexible">
+          <Image
+            styleName="medium-wide"
+            source={% raw %}{{{% endraw %} uri: restaurant.image.url  }}
+          />
+          <View styleName="content">
+            <Subtitle numberOfLines={3}>{restaurant.name}</Subtitle>
+            <View styleName="horizontal">
+              <Caption styleName="collapsible" numberOfLines={2}>{restaurant.address}</Caption>
+            </View>
+          </View>
+        </Card>
+      </TouchableOpacity>
     );
-  }
+  });
+
+  return (
+    <GridRow columns={2}>
+      {cellViews}
+    </GridRow>
+  );
+}
+
+render() {
+  // Group the restaurants into rows with 2 columns, except for the
+  // first article. The first article is treated as a featured article
+  let isFirstArticle = true;
+  const groupedData = GridRow.groupByRows(this.state.restaurants, 2, () => {
```

Comment:
We should avoid `let` in examples. You can extract this whole logic into a separate function that calculates `isFirstArticle`.